### PR TITLE
Update to latest discovery

### PR DESCRIPTION
**Pull request by AgeManning**
This update brings a few changes. 

Specifically: 
- More accessibility for NAT'd peers to find and discover peers
- Improved enr-auto-updates for dual-stack nodes

### CODE REVIEW

**Review by pawanjay176 — APPROVED** [review]
This looks good, but I can't test it as I don't have an ipv6 connection at home.
Have you tested auto-enr with ipv6?

**Review by jxs — APPROVED** [review]

**Comment by AgeManning** [comment]
Yeah. I've been testing with ipv6 using discv5-cli. 

There are probably some other changes we might need to introduce for ipv6 however. I've not quite got to them yet.

**Comment by AgeManning** [comment]
I've made a new commit. Moves to a whole new discovery. 

I think this current version should be fine to support ipv6 and potentially dual-stack by default

File: beacon_node/lighthouse_network/src/config.rs [hunk 1]
```diff
@@ -305,12 +305,12 @@ impl Default for Config {
...
             .enable_packet_filter()
             .session_cache_capacity(5000)
-            .request_timeout(Duration::from_secs(1))
+            .request_timeout(Duration::from_secs(2))
             .query_peer_timeout(Duration::from_secs(2))
             .query_timeout(Duration::from_secs(30))
             .request_retries(1)
             .enr_peer_update_min(10)
-            .query_parallelism(5)
+            .query_parallelism(8)
```
Comment (pawanjay176) [inline beacon_node/lighthouse_network/src/config.rs:313]: Why is this bumped?

**Comment by michaelsproul** [comment]
I've pushed a commit to minimise the update to `Cargo.lock`, seeing as there's something nasty lurking in the dep tree that we haven't found yet. See:

- https://github.com/sigp/lighthouse/pull/6554

File: beacon_node/lighthouse_network/src/config.rs (hunk 1 above)
Comment (AgeManning) [reply beacon_node/lighthouse_network/src/config.rs:313]: Some simulations or experiments were saying discovery can run faster just by increasing the parallelism. 

I was looking through the configs, and while I was there, figured I'd increase. Happy to leave it as is, if its a concern. 

Its just that I've been meaning to mess with it for a while, so figure in the update, might as well do it here.

**Comment by michaelsproul** [comment]
@mergify queue

**Pull request merged**